kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/auth/AwsContextCreator.scala

```diff
@@ -44,19 +55,27 @@ object AwsContextCreator {
 
   }
 
-  private def getAuthModeFn(awsConfig: S3Config): (String, String) =
-
+  private def credentialsSupplier(awsConfig: S3Config): Supplier[Credentials] = {
     awsConfig.authMode match {
-      case AuthMode.Credentials => credentialsConfigFn(awsConfig: S3Config)
-      case _ => throwErrorConfigFn(awsConfig.authMode)
+      case AuthMode.Credentials => credentialsFromConfig(awsConfig: S3Config)
+      case _ => credentialsFromDefaultChain()
     }
+  }
 
-  private def credentialsConfigFn = (awsConfig: S3Config) => (
-    awsConfig.accessKey,
-    awsConfig.secretKey
+  private def credentialsFromConfig(awsConfig: S3Config): Supplier[Credentials] = () => new Credentials(
+    awsConfig.accessKey.getOrElse(throw new IllegalArgumentException(missingCredentialsError)),
```

Comment:
The config defines  empty values as default.  so when someone uses the config we should check for empty values as the result of getOrElse no?